database/src/desktop/core/repo.h

```diff
@@ -158,6 +158,10 @@ class Repo : public connection::PersistentConnectionEventHandler {
 
   void UpdateInfo(const std::string& key, const Variant& value);
 
+  // Callback function used to listen for App Check token changes, and updating
+  // the connection with them.
+  static void OnAppCheckTokenChanged(std::string token, void* context);
```

Comment:
Use `const std::string&` to reduce string copy?
Is there a particular reason this cannot be used for a FunctionRegistry callback?